issuetracker/src/main/java/org/openjdk/skara/issuetracker/jira/JiraProject.java

```diff
@@ -484,49 +470,65 @@ public List<IssueTrackerIssue> issues(ZonedDateTime updatedAfter) {
-        var issues = request.get("search")
-                .param("jql", jql)
-                .param("maxResults", "1")
-                .execute();
-        var issuesArray = issues.get("issues").asArray();
-        if (issuesArray.isEmpty()) {
-            return Optional.empty();
-        } else {
-            var json = issuesArray.get(0);
-            return Optional.of(new JiraIssue(this, generateIssueRequest(json), json));
-        }
+        var issues = search("project = " + projectName + " ORDER BY updated DESC", 1);
+        return issues.isEmpty() ? Optional.empty() : Optional.of(issues.get(0));
     }
 
     private String toTimeString(ZonedDateTime time) {
         var timeZoned = time.withZoneSameInstant(jiraHost.timeZone());
         return timeZoned.format(DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm"));
     }
 
-    private List<IssueTrackerIssue> queryIssues(String jql) {
+    private List<IssueTrackerIssue> search(String jql) {
+        return search(jql, -1);
+    }
+
+    private List<IssueTrackerIssue> search(String jql, int limit) {
+        if (limit == 0) {
+            return List.of();
+        }
+
+        var req = request.get("search").param("jql", jql);
+        if (limit > 0) {
+            req = req.param("maxResults", Integer.toString(limit));
+        }
+        var res = req.execute();
+
+        // Due to pagination the same issue can be returned multiple times,
+        // for example if an issue is updated _after_ the REST call to the
+        // "search" endpoint. Then it might be included in one of  the next
+        // pages of results, even though it was also present in the first page.
+        //
+        // Handle this case gracefuily by prefering the most recent version of
+        // the issue (the one encountered last), this is implemented by
+        // utilizing a HashMap.
         var ret = new HashMap<String, IssueTrackerIssue>();
+        var issues = res.get("issues").asArray();
         int count = 0;
-        var issues = request.get("search")
-                .param("jql", jql)
-                .execute();
-        var startAt = 0;
-        while (issues.get("issues").asArray().size() > 0) {
-            for (var issue : issues.get("issues").asArray()) {
+        while (issues.size() > 0) {
+            for (var issue : issues) {
                 ret.put(JiraIssue.id(issue), new JiraIssue(this, generateIssueRequest(issue), issue));
                 count++;
+                if (count == limit) {
+                    break;
+                }
             }
 
-            if (count < issues.get("total").asInt()) {
-                startAt += issues.get("issues").asArray().size();
-                issues = request.get("search")
-                        .param("jql", jql)
-                        .param("startAt", String.valueOf(startAt))
-                        .execute();
+            if (count < issues.get("total").asInt() && count < limit) {
```

Comment:
If `limit == -1` then we will never try to request more pages. Is that intended?